### ISSUE TITLE
Kape not expanding target Paths

### ISSUE THREAD

**Issue by mcvic1rj** [EricZimmerman/KapeFiles#664]
**KAPE version**

v1.2.0.0 and v1.1.0.1

**System Information**
VMware Fusion VM running Windows 10 Education build 19044

**Describe the bug**

Kape Does not appear to be expanding wildcards in a targets Path parameter. Based on ProcMon, Kape does not appear to even query for the directory.

**To Reproduce**
Using a pared down version of the TeamViewer target with the command `.\kape.exe --tsource C --tdest c:\kapes\ --tflush --target TeamViewerLogs --debug --gui`

```yaml
Description: TeamViewer Logs
Author: Hadar Yudovich
Version: 1.1
Id: 6f2cd531-1f4b-4f0b-aa96-2426621b0a14
RecreateDirectories: true
Targets:
    -
        Name: TeamViewer Application Logs wildcard
        Category: ApplicationLogs
        Path: C:\Program Files*\TeamViewer\
        FileMask: 'TeamViewer*_Logfile*'
        Comment: "Includes TeamViewer<version>_Logfile.log and TeamViewer<version>_Logfile_OLD.log"
```

**Expected behavior**

Expansion of the `Path` parameter to include `C:\Program Files\` and `C:\Program Files (x86)\` 

**Screenshots**

<img width="1153" alt="image" src="https://user-images.githubusercontent.com/10363041/178005689-5e5736a3-f580-468a-a304-339552c4bf4a.png">

**Additional context**
This does not appear to happen with other folders/paths like `C:\Users\*\`

Switching the path to `C:\Program File*\TeamViewer\` to possibly rule out an expansion issue results in the same issue.

**Comment by EricZimmerman** [comment]
enable debug and or trace and see what the logs tell you.

**Comment by EricZimmerman** [comment]
also i would consider just dropping the * in the path and adding the x86 and normal path as 2 entries. its just easier for people to understand

**Comment by EricZimmerman** [comment]
i would not expect the one with C:\z\to find anything at all

what does a path of

Path: "C:\Program Files\TeamViewer\",

do?

**Comment by EricZimmerman** [comment]
i think this approach

Targets:
    -
        Name: TeamViewer Application Logs
        Category: ApplicationLogs
        Path: C:\Program Files\TeamViewer\
        FileMask: 'TeamViewer*_Logfile*'
        Comment: "Includes TeamViewer<version>_Logfile.log and TeamViewer<version>_Logfile_OLD.log"
    - 
        Name: TeamViewer Application Logs x86
        Category: ApplicationLogs
        Path: C:\Program Files (x86)\TeamViewer\
        FileMask: 'TeamViewer*_Logfile*'
        Comment: "Includes TeamViewer<version>_Logfile.log and TeamViewer<version>_Logfile_OLD.log"
        
should do the trick.

i dont know of any other targets use something like 'program files*' and i can see about making it behave that way, but that was not the initial goal.

i suppose you can do

Path: C:\*\TeamViewer\

as well, but listing the 2 paths explicitly is better IMO

**Comment by EricZimmerman** [comment]
this is news to me. @AndrewRathbun can you take a look and see if we can update/determine when/why this started, assuming it ever worked?

**Comment by AndrewRathbun** [comment]
I thought it always worked but never noticed an issue until now. I can remediate those ones listed above so we're safe.

**Comment by mcvic1rj** [comment]
After additional testing, it seems like this issue can be clarified to the following:
When running KAPE from a mounted volume and targeting that SAME mounted volume, wild cards in the first directory after the mount letter are not expanded/searched.
Examples: 
Kape path: `c:\users\ryan\downloads\kape\kape.exe`
Target source: `c:`
Ends with no files found.

Working:
Kape path: `v:\kape\kape.exe`
Target source: `c:`
Ends with 1 file found and copied (expected)



Additionally a following issue was found that may be related, and happens regardless of where kape is run:
Wild cards in the first folder after the root drive are not expanded/lead to a terminating error.
Examples: `C:\*\` `C:\Window*`
Error Message:
```
2022-07-14 12:56:57.2362 | T | Target details: {
	Name: TeamViewer Application Logs,
	Category: ApplicationLogs,
	Path: "C:\*\TeamViewer\",
	Recursive: False,
	AlwaysAddToQueue: False,
	FileMask: TeamViewer*_Logfile*,
	Comment: Includes TeamViewer<version>_Logfile.log and TeamViewer<version>_Logfile_OLD.log
}
2022-07-14 12:56:57.2518 | D | Expanding directory: 'C:\*\TeamViewer\'
2022-07-14 12:56:57.2518 | T | START dir: C:\*\TeamViewer\
2022-07-14 12:56:57.7685 | F | Object reference not set to an instance of an object.:    at kape.Classes.Helpers.ResolveReparsePoints(String dir)
   at kape.Classes.Helpers.ExpandDirectory(String dir)
   at kape.Program.ExpandTarget(Target target)
   at kape.Program.ProcessTarget(Target target, String startDirectory)
   at kape.Program.LoadTargetFile(TargetCollection targetFile)
   at kape.Program.Main(String[] args)
2022-07-14 12:56:57.7990 | W | Unhandled exception! Attempting to write current Console Log to 'C:\Users\ryan\Downloads\KAPE\2022-07-14T16_56_54_8926204_ConsoleLog.txt'
```

**Comment by EricZimmerman** [comment]
this is fixed in 1.3. @AndrewRathbun ill send you a build to test/verify

no idea WHY it happens, but when kape and the tsource are the same, the currentdirectory was being set to the KAPE directory vs the root of tsource. 

i now just explicitly set the current directory to the base directory to search and things now work regardless

this ONLY happened when kape was run from the same drive as tsource.

![image](https://user-images.githubusercontent.com/4265254/179357735-488200e7-2af2-43de-ae79-ecdd8712283a.png)

**Comment by AndrewRathbun** [comment]
> this is fixed in 1.3. @AndrewRathbun ill send you a build to test/verify
> 
> no idea WHY it happens, but when kape and the tsource are the same, the currentdirectory was being set to the KAPE directory vs the root of tsource. 
> 
> i now just explicitly set the current directory to the base directory to search and things now work regardless
> 
> this ONLY happened when kape was run from the same drive as tsource.
> 
> ![image](https://user-images.githubusercontent.com/4265254/179357735-488200e7-2af2-43de-ae79-ecdd8712283a.png)
> 

Good stuff, thank you @EricZimmerman. Great find!

**Comment by mcvic1rj** [comment]
Also verified this issue was resolved in the beta build. Thank you for the great tools @EricZimmerman !
